packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -251,17 +251,54 @@ export interface FieldProps {
 	readonly key?: string;
 }
 
+/**
+ * This function exposes the private constructor for {@link FieldSchema} for public users.
+ * @public
+ */
+export const makeFieldSchema = <
+	Kind extends FieldKind = FieldKind,
+	Types extends ImplicitAllowedTypes = ImplicitAllowedTypes,
+>(
+	kind: Kind,
+	allowedTypes: Types,
+	props?: FieldProps,
+) => create(kind, allowedTypes, props);
+
+/**
+ * Package internal construction API.
+ * Use {@link makeFieldSchema} to create an instance of this type instead.
+ */
+let create: <
```

Comment:
since there are a bunch of types in this file, I don't think "create" is a sufficiently descriptive name. "createFieldSchema" would be better. Also note that while its a rather arbitrary choice, we tend to use "create"
 and not "make" in factory function names.

Also since we won't be making this public, I think its fine just export this and not provide a wrapper function around it (which I assume you did to hide the mutability).

Note that if you do want to keep the wrapper function, it would probably be better to place it after the FieldSchema class in this file to prevent it from being invoked before the function its calling is assigned (it won't really help much, but its very slightly safer)

Reply:
Removed wrapper function, renamed `create` function to `createFieldSchema` and exported